test/selenium/test_navigation.py

```diff
@@ -0,0 +1,31 @@
+from util import ExtensionTestCase
+
+kittens_url = 'http://freerangekitten.com/'
+
+mixed_url = 'http://very.badssl.com/'
+mixed_selector = '#content > div > form > fieldset > button'
+mixed_success_text = 'Welcome to the form submission page.'
+
+
+class TestNavigation(ExtensionTestCase):
+    def test_redirect(self):
+        self.driver.get(kittens_url)
+        self.assertTrue(self.driver.current_url.startswith('https'))
+
+    def test_no_redirect_when_disabled(self):
+        self.toggle_disabled()
+        self.driver.get(kittens_url)
+        self.assertEqual(self.driver.current_url, kittens_url)  # not https
+
+    def test_httpnowhere_blocks(self):
+        self.toggle_http_nowhere()
+        self.driver.get(mixed_url)
+        self.assertTrue(self.driver.current_url.startswith('https'))
+        self.query_selector(mixed_selector).click()
+        self.assertFalse(mixed_success_text in self.query_selector('body').text)
+
+    def test_mixed_content_not_blocked(self):
+        self.driver.get(mixed_url)
+        self.assertTrue(self.driver.current_url.startswith('https'))
+        self.query_selector(mixed_selector).click()
+        self.assertTrue(mixed_success_text in self.query_selector('body').text)
```

Comment:
@Hainish what else should I add here? We can probably use URLs from badssl.com

Reply:
This looks fine for now, let's look at this again when there's time

Reply:
Add js disabled here

Reply:
add testcase for if cookies disabled persisting settings works

check test/readme.md todo